File: apps/motoboy/app/src/test/kotlin/com/drogsantoantonio/farmadelivery/presentation/ui/screens/DeliveryPriorityLabelsTest.kt
package com.drogsantoantonio.farmadelivery.presentation.ui.screens

import kotlin.test.Test
import kotlin.test.assertEquals

class DeliveryPriorityLabelsTest {
  @Test
  fun `maps known delivery priorities to readable labels`() {
    assertEquals("Normal", deliveryPriorityLabel("NORMAL"))
    assertEquals("Urgente", deliveryPriorityLabel("URGENTE"))
    assertEquals("Retorno", deliveryPriorityLabel("RETORNO"))
  }

  @Test
  fun `formats unknown delivery priorities without raw underscores`() {
    assertEquals("Muito urgente", deliveryPriorityLabel("MUITO_URGENTE"))
  }

  @Test
  fun `uses a safe fallback for blank delivery priorities`() {
    assertEquals("Prioridade desconhecida", deliveryPriorityLabel("   "))
  }
}

File: apps/motoboy-pwa/src/deliveryPriorityLabels.test.ts
import { describe, it } from "node:test";
import assert from "node:assert/strict";
import { deliveryPriorityLabel } from "./deliveryPriorityLabels";

describe("deliveryPriorityLabel", () => {
  it("maps known delivery priorities to readable labels", () => {
    assert.equal(deliveryPriorityLabel("NORMAL"), "Normal");
    assert.equal(deliveryPriorityLabel("URGENTE"), "Urgente");
    assert.equal(deliveryPriorityLabel("RETORNO"), "Retorno");
  });

  it("formats unknown delivery priorities without raw underscores", () => {
    assert.equal(deliveryPriorityLabel("MUITO_URGENTE"), "Muito urgente");
  });

  it("uses a safe fallback for blank delivery priorities", () => {
    assert.equal(deliveryPriorityLabel("   "), "Prioridade desconhecida");
  });
});
